src/modules/auth_arnacon/auth_arnacon.c

```diff
@@ -0,0 +1,340 @@
+#include "../../modules/auth/api.h"
+
+#include "auth_arnacon.h"
+#include "arnacon_core.h"
+
+MODULE_VERSION
+
+/* Module parameters */
+static char *arnacon_ens_registry_address = "0x16742E546bF92118F7dfdbEF5170E44C47ae254b";
+static char *arnacon_rpc_url = "https://polygon-rpc.com";
+static int arnacon_signature_timeout = 30;
+static int arnacon_debug_mode = 0;
+
+/* Forward declarations */
+static int mod_init(void);
+static int child_init(int rank);
+static void destroy(void);
+static int arnacon_authenticate_user(struct sip_msg *msg, char *ens_name_param, char *x_data_param, char *x_sign_param);
+static int arnacon_user_exists(struct sip_msg *msg, char *ens_name_param, char *str2);
+
+/* Module interface */
+static cmd_export_t cmds[] = {
+    {"arnacon_authenticate", (cmd_function)arnacon_authenticate_user, 3, 0, 0, REQUEST_ROUTE},
+    {"arnacon_user_exists", (cmd_function)arnacon_user_exists, 1, 0, 0, REQUEST_ROUTE},
+    {0, 0, 0, 0, 0, 0}
+};
+
+static param_export_t params[] = {
+    {"ens_registry_address", PARAM_STRING, &arnacon_ens_registry_address},
+    {"rpc_url", PARAM_STRING, &arnacon_rpc_url},
+    {"signature_timeout", PARAM_INT, &arnacon_signature_timeout},
+    {"debug_mode", PARAM_INT, &arnacon_debug_mode},
+    {0, 0, 0}
+};
+
+struct module_exports exports = {
+    "auth_arnacon",    /* module name */
+    DEFAULT_DLFLAGS,   /* dlopen flags */
+    cmds,              /* exported functions */
+    params,            /* exported parameters */
+    0,                 /* exported RPC functions */
+    0,                 /* exported pseudo-variables */
+    0,                 /* response function */
+    mod_init,          /* module initialization function */
+    child_init,        /* per child init function */
+    destroy            /* destroy function */
+};
+
+/**
+ * Module initialization function
+ */
+static int mod_init(void)
+{
+    LM_INFO("Initializing...");
+    
+    if (arnacon_debug_mode) {
+        LM_INFO("Debug mode enabled");
+        LM_INFO("ENS Registry: %s", arnacon_ens_registry_address);
+        LM_INFO("RPC URL: %s", arnacon_rpc_url);
+        LM_INFO("Signature timeout: %d seconds", arnacon_signature_timeout);
+        LM_INFO("Name Wrapper detection: Dynamic (no configuration needed)");
+    }
+    
+    /* Initialize curl globally */
+    if (arnacon_core_init() != 0) {
+        LM_ERR("Failed to initialize core");
+        return -1;
+    }
+    
+    LM_INFO("Initialized successfully");
+    return 0;
+}
+
+/**
+ * Child initialization function
+ */
+static int child_init(int rank)
+{
+    LM_DBG("Child process %d initialized", rank);
+    return 0;
+}
+
+/**
+ * Module destroy function
+ */
+static void destroy(void)
+{
+    LM_INFO("Shutting down");
+    arnacon_core_cleanup();
+}
+
+/**
+ * Extract string parameter from Kamailio function call
+ * Handles both literal strings and pseudo-variables
+ */
+static int get_str_param(struct sip_msg *msg, char *param, str *result)
```

Comment:
Why is this necessary? If I am not mistaken, there are functions to just print a string from the cfg, regardless if its a literal string or contains PVs.